searx/engines/voidlinux.py

```diff
@@ -22,30 +22,63 @@
 base_url = "https://xq-api.voidlinux.org"
 pkg_repo_url = "https://github.com/void-linux/void-packages"
 
+void_arch = 'x86_64'
+"""Default architecture to search for.  For valid values see :py:obj:`ARCH_RE`"""
+
+ARCH_RE = re.compile('aarch64-musl|armv6l-musl|armv7l-musl|x86_64-musl|aarch64|armv6l|armv7l|i686|x86_64')
+"""Regular expresion that match a architecture in the query string."""
+
 
 def request(query, params):
-    params['url'] = f"{base_url}/v1/query/x86_64?q={quote_plus(query)}"
+    arch_path = ARCH_RE.search(query)
+    if arch_path:
+        arch_path = arch_path.group(0)
+        query = query.replace(arch_path, '').strip()
+    else:
+        arch_path = void_arch
+
+    params['url'] = f"{base_url}/v1/query/{arch_path}?q={quote_plus(query)}"
     return params
 
 
 def response(resp):
-    results = []
+    """
+    At Void Linux, several packages sometimes share the same source code
+    (template) and therefore also have the same URL.  Results with identical
+    URLs are merged as one result for SearXNG.
+    """
 
+    packages = {}
     for result in resp.json()['data']:
+
         # 32bit and dbg packages don't have their own package templates
         github_slug = re.sub(r"-(32bit|dbg)$", "", result['name'])
+        pkg_url = f"{pkg_repo_url}/tree/master/srcpkgs/{github_slug}"
 
-        results.append(
+        pkg_list = packages.get(pkg_url, [])
+        pkg_list.append(
             {
-                'template': 'packages.html',
-                'allow_duplicated_urls': True,
-                'url': f"{pkg_repo_url}/tree/master/srcpkgs/{github_slug}",
                 'title': result['name'],
                 'content': f"{result['short_desc']} - {humanize_bytes(result['filename_size'])}",
                 'package_name': result['name'],
                 'version': f"v{result['version']}_{result['revision']}",
-                'tags': [result['repository']],
+                'tags': result['repository'],
             }
         )
+        packages[pkg_url] = pkg_list
 
+    results = []
+    for pkg_url, pkg_list in packages.items():
+
+        results.append(
+            {
+                'url': pkg_url,
+                'template': 'packages.html',
+                'title': ' | '.join(x['title'] for x in pkg_list),
+                'content': ' | '.join(x['content'] for x in pkg_list),
+                'package_name': ' | '.join(x['package_name'] for x in pkg_list),
+                'version': ' | '.join(x['version'] for x in pkg_list),
```

Comment:
Packages that are `dbg` or `32bit` always have the same version as the main package, so I think we can just use `pkg_list[0]['version']` here.